services/web/server/src/simcore_service_webserver/director_v2_core_dynamic_services.py

```diff
@@ -0,0 +1,218 @@
+        app, "GET", backend_url, expected_status=web.HTTPOk
+    )
+
+    assert isinstance(service_state, dict)  # nosec
+    return service_state
+
+
+@log_decorator(logger=log)
+async def run_dynamic_service(
+    app: web.Application,
+    user_id: PositiveInt,
+    project_id: str,
+    service_key: str,
+    service_version: str,
+    service_uuid: str,
+    request_dns: str,
+    request_scheme: str,
+    service_resources: ServiceResourcesDict,
+) -> DataType:
+    """
+    Requests to run (i.e. create and start) a dynamic service:
+    - legacy services request is redirected to `director-v0`
+    - dynamic-sidecar `director-v2` will handle the request
+    """
+    data = {
+        "user_id": user_id,
+        "project_id": project_id,
+        "key": service_key,
+        "version": service_version,
+        "node_uuid": service_uuid,
+        "basepath": f"/x/{service_uuid}",
+        "service_resources": ServiceResourcesDictHelpers.create_jsonable(
+            service_resources
+        ),
+    }
+
+    headers = {
+        "X-Dynamic-Sidecar-Request-DNS": request_dns,
+        "X-Dynamic-Sidecar-Request-Scheme": request_scheme,
+    }
+
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    started_service = await request_director_v2(
+        app,
+        "POST",
+        url=settings.base_url / "dynamic_services",
+        data=data,
+        headers=headers,
+        expected_status=web.HTTPCreated,
+    )
+
+    assert isinstance(started_service, dict)  # nosec
+    return started_service
+
+
+@log_decorator(logger=log)
+async def stop_dynamic_service(
+    app: web.Application, service_uuid: str, save_state: bool = True
+) -> None:
+    # stopping a service can take a lot of time
+    # bumping the stop command timeout to 1 hour
+    # this will allow to sava bigger datasets from the services
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    await request_director_v2(
+        app,
+        "DELETE",
+        url=(settings.base_url / f"dynamic_services/{service_uuid}").update_query(
+            save_state="true" if save_state else "false",
+        ),
+        expected_status=web.HTTPNoContent,
+        timeout=settings.DIRECTOR_V2_STOP_SERVICE_TIMEOUT,
+    )
+
+
+@log_decorator(logger=log)
+async def stop_dynamic_services_in_project(
+    app: web.Application,
+    user_id: Optional[PositiveInt] = None,
+    project_id: Optional[str] = None,
+    save_state: bool = True,
+) -> None:
+    """Stops all services of either project_id or user_id in concurrently"""
+    running_dynamic_services = await get_dynamic_services(
+        app, user_id=user_id, project_id=project_id
+    )
+
+    services_to_stop = [
+        stop_dynamic_service(
+            app=app, service_uuid=service["service_uuid"], save_state=save_state
+        )
+        for service in running_dynamic_services
+    ]
+    await logged_gather(*services_to_stop)
+
+
+# FIXME: ANE please unduplicate the 2 following calls!!!!
```

Comment:
Question: How do you simcore devs usually handle this stuff, what is the standard operating procedure? Will a FIXME in the code be sufficient for now, even if it is forgotten eventually, or do you additionally open an issue?

Reply:
see new comment and https://github.com/ITISFoundation/osparc-simcore/issues/3191